nifi-docs/src/main/asciidoc/administration-guide.adoc

```diff
@@ -489,6 +489,28 @@ To enable authentication via Apache Knox the following properties must be config
 this listing. The audience that is populated in the token can be configured in Knox.
 |==================================================================================================================================================
 
+[[json_web_token]]
+=== JSON Web Tokens
+
+NiFi uses JSON Web Tokens to provide authenticated access after the initial login process. Generated JSON Web Tokens include the authenticated user identity
+as well as the issuer and expiration from the configured Login Identity Provider.
+
+NiFi uses generated RSA Key Pairs with a key size of 4096 bits to support the `RS512` algorithm for JSON Web Signatures. The system stores RSA
```

Comment:
The current implementation uses a hard-coded setting for the key pair size to avoid potential misconfiguration, however, it may be more than necessary in some instances. Given the other resource constraints on a running system, this does not seem to be too expensive for most deployments.  An alternative could be to make this configurable, but there is a tradeoff between having too many settings and sensible defaults. What do you think?

Reply:
I see the recent industry movement to 3072-bit minimum RSA key lengths.  4096 is probably a good choice to future proof this implementation.  I have concerns about performance of key generation at startup on lower-end devices, but this is probably not a valid concern for NiFi.

https://knowledge.digicert.com/alerts/code-signing-new-minimum-rsa-keysize.html

Reply:
Thanks for the follow up and background reference, that is helpful to know.